src/FixedDataTable.js

```diff
@@ -463,52 +465,70 @@ class FixedDataTable extends React.Component {
     this._didScrollStopSync();
   }
 
-  _shouldHandleTouchX = (/*number*/ delta) /*boolean*/ =>
-    this.props.touchScrollEnabled && this._shouldHandleWheelX(delta)
+  _shouldHandleWheelX = (/*number*/ delta) /*boolean*/ => {
+    return this.props.touchScrollEnabled && this._shouldHandleScrollX(delta);
+  }
+
+  _shouldHandleWheelY = (/*number*/ delta) /*boolean*/ => {
+    return this.props.touchScrollEnabled && this._shouldHandleScrollY(delta);
+  }
 
-  _shouldHandleTouchY = (/*number*/ delta) /*boolean*/ =>
-    this.props.touchScrollEnabled && this._shouldHandleWheelY(delta)
+  _shouldHandleTouchX = (/*number*/ delta) /*boolean*/ => {
+    const { threshold } = configSelector(this.props).touchConfig;
+    return this.props.touchScrollEnabled && this._shouldHandleScrollX(delta, threshold);
+  }
 
-  _shouldHandleWheelX = (/*number*/ delta) /*boolean*/ => {
+  _shouldHandleTouchY = (/*number*/ delta) /*boolean*/ => {
+    const { threshold } = configSelector(this.props).touchConfig;
+    return this.props.touchScrollEnabled && this._shouldHandleScrollY(delta, threshold);
+  }
+
+  _shouldHandleScrollX = (/*number*/ delta, threshold = 0) /*boolean*/ => {
     const { maxScrollX, scrollFlags, scrollX } = this.props;
     const { overflowX } = scrollFlags;
 
-    if (overflowX === 'hidden') {
+    if (overflowX === 'hidden' || delta == 0) {
       return false;
     }
 
-    delta = Math.round(delta);
-    if (delta === 0) {
```

Comment:
If we round delta and check it for zero, very small values will get ignored (most noticeable in touch scroll).